tool_helper_utils/__init__.py

```diff
@@ -2,12 +2,30 @@
 import sys
 
 
+class AbstractBaseFactoryFactory:
+    def build():
+        return None
+
+
+class FactoryFactory(AbstractBaseFactoryFactory):
+    def __init__(self, factory):
+        self.factory = factory
+
+    def build(self):
+        return self.factory.get_bad_idea()
+       
+
+class FactoryBadIdea:
+    def get_bad_idea(self):
+        return "factory"
```

Comment:
This is a lot of classes for a single source file. We should think about factoryng them out into a separate file.